File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = -1
    # 방법
    st = []
    l = len(s)
    for i in s:
        #넣는다
        st.append(i)
        #뺀다
        while len(st) >= 2 and st[-2]==st[-1]:
            st.pop()
            st.pop()
    
    return 1 if len(st)==0 else 0

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s) {
  const stack = [];

  for (let i = 0; i < s.length; i++) {
    const curr = s.charAt(i);
    if (stack[stack.length - 1] === curr) {
      stack.pop();
    } else {
      stack.push(curr);
    }
  }

  return stack.length === 0 ? 1 : 0;
}